frontend/src/metabase/common/components/QueryColumnPicker/QueryColumnPicker.unit.spec.tsx

```diff
@@ -61,12 +82,50 @@ describe("QueryColumnPicker", () => {
     ).toBeInTheDocument();
   });
 
-  it("should allow picking a column", async () => {
+  it("should allow picking a column", () => {
     const { sampleColumn, sampleColumnInfo, onSelect, onClose } = setup();
 
     userEvent.click(screen.getByText(sampleColumnInfo.displayName));
 
     expect(onSelect).toHaveBeenCalledWith(sampleColumn);
     expect(onClose).toHaveBeenCalled();
   });
+
+  it("should highlight column used in a given clause", () => {
```

Comment:
Also binning tests?

Reply:
@ranquild it's a bit tricky to test for `QueryColumnPicker`

If we want to test bucketing (binning/temporal), we'd most likely want to assert that the new column's display name mentions a bucket:

```js
// pseudo-code
userEvent.click(screen.getByTestId("temporal-bucket-year"));

const nextColumn = getNextColumnFromMockCallback()
const nextColumnInfo = Lib.displayInfo(query, stageIndex, nextColumn)
expect(nextColumnInfo.displayName).toBe("Created At: Year")
```

The tricky part is that `Lib.displayInfo(query, stageIndex, nextColumn)` returns bucket-aware info only when this column is actually used in a query. That means we have to e.g. make the `QueryColumnPicker` test use breakouts or whatever. I'm not sure if we should make the column picker's test suite depend on breakouts or other clauses. FWIW, bucketing is covered with tests at [`BreakoutStep.unit.spec.tsx` (link)](https://github.com/metabase/metabase/blob/74160e9f01d7bd52047bcb8afa4566c3cb944e7a/frontend/src/metabase/query_builder/components/notebook/steps/BreakoutStep/BreakoutStep.unit.spec.tsx#L150)

> The tricky part is that `Lib.displayInfo(query, stageIndex, nextColumn)` returns bucket-aware info only when this column is actually used in a query

I'm now not sure if this is expected, [asked internally](https://metaboat.slack.com/archives/C04DN5VRQM6/p1686754424678519)